extensions/ql-vscode/src/contextual/templateProvider.ts

```diff
@@ -199,6 +211,31 @@ export class TemplatePrintAstProvider {
         zippedArchive.pathWithinSourceArchive
     };
 
+    // The AST viewer queries now live within the standard library packs.
+    // This simplifies distribution (you don't need the standard query pack to use the AST viewer),
+    // but if the library pack doesn't have a lockfile, we won't be able to find
+    // other pack dependencies of the library pack.
+
+    // Work out the enclosing pack.
+    const packContents = await this.cli.packPacklist(query, false);
+    const packFilePath = packContents.find((p) => ['codeql-pack.yml', 'qlpack.yml'].includes(path.basename(p)));
+    if (packFilePath === undefined) {
+      // Should not happen; we already resolved this query.
+      throw new Error(`Could not find a CodeQL pack file for the pack enclosing the query ${query}`);
+    }
+    const packPath = path.dirname(packFilePath);
+    const lockFilePath = packContents.find((p) => ['codeql-pack.lock.yml', 'qlpack.lock.yml'].includes(path.basename(p)));
+    if (!lockFilePath) {
+      // No lock file, likely because this library pack is in the package cache.
+      // Create a lock file so that we can resolve dependencies and library path
+      // for the AST query.
+      void logger.log(`Library pack ${packPath} is missing a lock file; creating a temporary lock file`);
+      await this.cli.packResolveDependencies(packPath, 'use-lock');
+      // Install dependencies.
+      void logger.log(`Installing package dependencies for library pack ${packPath}`);
+      await this.cli.packInstall(packPath);
```

Comment:
Likely you need to clear the query server cache before running. That probably explains why it wasn't working for you earlier. The cache is cleared every 2 seconds or so. 

```suggestion
      await this.cli.clearCache();
      await this.cli.packInstall(packPath);
```

If you don't clear the cache, then it will be using an old qlpack instance that will not know about the new lock file.

Reply:
I had no idea. Thank you!